source/common/router/router.cc

```diff
@@ -558,6 +558,31 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
     return Http::FilterHeadersStatus::StopIteration;
   }
 
+  // Support DROP_OVERLOAD config from control plane to drop certain percentage of traffic.
+  if (cluster->dropOverload().value()) {
```

Comment:
The `decodeHeaders` function has grown to be very long. Can you refactor this logic into a separate method, please? This should make decodeHeaders more readable.

Reply:
done!